### ISSUE TITLE
Potential Causes for High Malignant Cell Proportion in Immune Cell Clusters

### ISSUE THREAD

**Issue by suhuanhou** [Jonyyqn/scMalignantFinder#13]
Thank you very much for sharing this excellent tool/work!

When I applied the tool, I noticed that the proportion of malignant cells among immune cells is also quite high. This seems biologically implausible, as immune cells are generally not expected to be malignant. I am curious whether this is due to a misannotation, a data processing issue, or something I misunderstood.

The input I provide is adata (the normalized data), with norm_type set to False.

<img width="2347" height="1155" alt="Image" src="https://github.com/user-attachments/assets/310b94fa-e175-4eb2-9a0e-9570b84701e2" />

**Comment by Jonyyqn** [comment]
> Thank you very much for sharing this excellent tool/work!
> 
> When I applied the tool, I noticed that the proportion of malignant cells among immune cells is also quite high. This seems biologically implausible, as immune cells are generally not expected to be malignant. I am curious whether this is due to a misannotation, a data processing issue, or something I misunderstood.
> 
> The input I provide is adata (the normalized data), with norm_type set to False.
> 
> <img alt="Image" width="2000" height="984" src="https://private-user-images.githubusercontent.com/51230435/626035257-310b94fa-e175-4eb2-9a0e-9570b84701e2.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODQ4ODQ1MTYsIm5iZiI6MTc4NDg4NDIxNiwicGF0aCI6Ii81MTIzMDQzNS82MjYwMzUyNTctMzEwYjk0ZmEtZTE3NS00ZWIyLTlhMGUtOTU3MGI4NDcwMWUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzI0VDA5MTAxNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhMDU3ZDIyMjg3MDUzMGViNmU4YWU3NzhjNGRkMDgzYmFiOTYzM2YxM2UxYzI3Yzk1YzZlZmViMzhjNDBiNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.WLaceTHPJBHWF0sZJc7rXm2Xgb7mSJtrcYeFcn8s4Mw">

Hi @suhuanhou ,

Thank you for reaching out and for your support of scMalignantFinder!

To clarify, scMalignantFinder is specifically designed and trained to distinguish malignant epithelial cells from normal/non-malignant epithelial cells.

Here are a few important details regarding the model's design and usage:

Training Data Scope: The model was trained exclusively on single-cell RNA sequencing datasets of epithelial cells. Consequently, it is not trained or intended to predict the "malignant/benign" status of non-epithelial lineage cells (such as immune, stromal, or endothelial cells).

Recommended Workflow: We strongly recommend subsetting your AnnData object to include only epithelial cells (e.g., after initial cell-type annotation) before passing it to scMalignantFinder. Running the tool on mixed non-epithelial populations will result in biological misinterpretations.

For more details on the model architecture, training data composition, and benchmark evaluations, please refer to our paper in Communications Biology:
👉 [Read the paper here](https://www.nature.com/articles/s42003-025-07942-y)

Hope this helps clear up the confusion! Please let us know if you run into any other questions.
